File: android/versioned-abis/expoview-abi42_0_0/src/main/java/abi42_0_0/expo/modules/sensors/services/GravitySensorService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package abi42_0_0.expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import abi42_0_0.org.unimodules.core.interfaces.InternalModule;

import abi42_0_0.expo.modules.interfaces.sensors.services.GravitySensorServiceInterface;

public class GravitySensorService extends SubscribableSensorService implements InternalModule, GravitySensorServiceInterface {
  public GravitySensorService(Context reactContext) {
    super(reactContext);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_GRAVITY;
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(GravitySensorServiceInterface.class);
  }
}

File: android/versioned-abis/expoview-abi47_0_0/src/main/java/abi47_0_0/expo/modules/sensors/services/GravitySensorService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package abi47_0_0.expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import abi47_0_0.expo.modules.interfaces.sensors.services.GravitySensorServiceInterface
import abi47_0_0.expo.modules.core.interfaces.InternalModule

class GravitySensorService(reactContext: Context?) : SubscribableSensorService(reactContext), InternalModule, GravitySensorServiceInterface {
  override val sensorType: Int = Sensor.TYPE_GRAVITY

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(GravitySensorServiceInterface::class.java)
  }
}
